Python/pywarpx/picmi.py

```diff
@@ -417,7 +419,11 @@ def initialize_inputs(self):
 
         # Maximum allowable size of each subdomain in the problem domain;
         #    this is used to decompose the domain for parallel calculations.
-        pywarpx.amr.max_grid_size = self.max_grid_size
+        if self.max_grid_size_x is None or self.max_grid_size_y is None:
```

Comment:
This logic is not necessary. The C++ code already handles the case where `max_grid_size` and `max_grid_size_[x,y,z]` are set. If both are set, it gives preference to the `[x,y,z]` values. You can just set them all.

Reply:
Thank you, Dave. Done in https://github.com/ECP-WarpX/WarpX/pull/2524/commits/61ed4655725f36e02df43993a7d9510a95674903.